pylint/checkers/refactoring/recommendation_checker.py

```diff
@@ -56,10 +57,15 @@ class RecommendationChecker(checkers.BaseChecker):
             "Formatting a regular string which could be a f-string",
             "consider-using-f-string",
             "Used when we detect a string that is being formatted with format() or % "
-            "which could potentially be a f-string. The use of f-strings is preferred.",
+            "which could potentially be a f-string. The use of f-strings is preferred. "
+            "Requires Python 3.6",
```

Comment:
I believe the check doesn't actually require Python 3.6, but that `py-version` is set to >= that. In that sense, this is misleading.

Same problem exists in R6103 docs (and ChangeLog and whatsnew entry for it). Fixed also in a few related places below.

```suggestion
            "Requires py-version >= 3.6",
```

Reply:
I see what you mean and how it can be misleading. However, I don't think using `py-version` would be better. Technically `f-strings` do require Python 3.6. Most users shouldn't need to set `py-version` as long as they run `pylint` with the oldest version that their project supports.

Another argument, if a project still supports Python 3.5, it's also correct to state that 3.6 is required.

Reply:
> Most users shouldn't need to set py-version as long as they run pylint with the oldest version that their project supports.

Yup. But is there any documentation that would hint or explicitly say this would be the recommended thing to do? How could users tell? Unless one is familiar with the tool internals, going with the latest Python would be arguably an equally good or even better choice on surface.

For the f-strings case the Python version vs py-version distinction is coincidentally not that relevant if pylint won't run with earlier than 3.6. But saying "Requires Python 3.8" for a feature that can be checked with pylint running with Python 3.6 perfectly well it wouldn't be so. I haven't actually checked if such a case exists currently (i.e. if one can get consider-using-assignment-expr checked and messages when running with 3.6), but talking based on a hunch that it could. For that case I don't think anything besides referring to "py-version >= 3.8" would be the right thing to do, and for the same reasons, ignoring the noted coincidence, I still think documenting in terms of py-version would be the better choice here as well.

Reply:
(Anyway, happy to see this went in, thanks a bunch for that! Doc matters such as the one being discussed here, while important, are a secondary concern.)

Reply:
> Yup. But is there any documentation that would hint or explicitly say this would be the recommended thing to do?

Not that I know of. That is certainly an area that needs improvement. See #5038

> How could users tell? Unless one is familiar with the tool internals, going with the latest Python would be arguably an equally good or even better choice on surface.

I generalized `py-version` exactly for that use case. For `pylint` and `astroid` we have a `pylint` pre-commit hook that uses the local `pylint` install. I'm using Python 3.10 at the moment, so every time pre-commit would fail for things I couldn't fix as `pylint` still supports Python 3.6.

> But saying "Requires Python 3.8" for a feature that can be checked with pylint running with Python 3.6 perfectly well it wouldn't be so. I haven't actually checked if such a case exists currently (i.e. if one can get consider-using-assignment-expr checked and messages when running with 3.6), but talking based on a hunch that it could. For that case I don't think anything besides referring to "py-version >= 3.8" would be the right thing to do, and for the same reasons, ignoring the noted coincidence, I still think documenting in terms of py-version would be the better choice here as well.

`consider-using-assignment-expr` could indeed be checked with pylint on `3.6`. However, I think of it differently. If a user uses `Python 3.6` for `pylint`, he/she most likely also still supports it for their project. (Otherwise they wouldn't be using 3.6 in the first place.) Thus by saying `Requires Python 3.8`, even a beginner should now that he/she needs to update their Python install first.

In contrast, using `py-version >= 3.8` might lead to a situation where they think they can just set and start using it without updating Python. Although (hopefully) not that common, I imaging that would be quite frustrating.

Reply:
> Otherwise they wouldn't be using 3.6 in the first place.

That is a fair point. I don't quite agree with the rest of the argumentation though, I think it trades potential beginner friendliness for inaccuracy, and for project like pylint, to me the latter is _vastly_ more important. But I'm happy to leave it for the pylint maintainers to decide, just wanted to raise the point and thought I'd say what I'd do if it was my call.

Reply:
Maybe we can be very specific ? It would probably prevent multiple stackoverflow question or pylint's issues. It's not like we have a limited number of characters available 😄 . ``"Requires a python interpreter >= 3.6 to work and the pylint option 'py-version' set to >= 3.6 to be enabled"`` ?

Reply:
Just a note, the former "to work" part in that wording would be ambiguous -- it's not clear from that alone whether it means the check would work, or f-strings would work. "f-strings require Python >= 3.6, and this check is enabled only if the pylint option py-version is set to >= 3.6" would be more accurate if you want to go that route.